src/homework1.py

```diff
@@ -0,0 +1,124 @@
+import sys
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.tree import DecisionTreeClassifier
+
+
+def main():
+    df = pd.read_csv(
+        "../src/iris.data",
+        names=[
+            "sepal length (cm)",
+            "sepal width (cm)",
+            "petal length (cm)",
+            "petal width (cm)",
+            "class",
+        ],
+    )
+
+    for i in df.columns[:4]:
+        print(i)
+        print("Mean: " + str(np.mean(df[i])))
+        print("Min: " + str(np.min(df[i])))
+        print("Max: " + str(np.max(df[i])))
+        print("Q1: " + str(np.quantile(df[i], 0.25)))
+        print("Q2: " + str(np.quantile(df[i], 0.50)))
+        print("Q3: " + str(np.quantile(df[i], 0.75)))
+        print()
+
+    fig = px.scatter(
+        df,
+        title="Scatter plot - Sepal Width vs Sepal Length",
+        x="sepal length (cm)",
+        y="sepal width (cm)",
+        color="class",
+    )
+    fig.show()
+
+    fig = px.violin(
```

Comment:
For plotting, some of them are color separated by the class, and some are not, since they are already separated in the violin format, I was wondering if you'd like to change it to different color by the class as well.

Reply:
Thanks for the idea, I have changed this and it will be in the next commit